rdt/transformers/CatTransformer.py

```diff
@@ -20,8 +20,11 @@ def fit_transform(self, col, col_meta, missing=True):
         """ Returns a tuple (transformed_table, new_table_meta) """
         out = pd.DataFrame()
         col_name = col_meta['name']
-        self.get_probability_map(col)
-        out[col_name] = col.apply(self.get_val)
+        # Make sure all nans are handled the same by replacing
+        # with -1
+        column = col.fillna(-1)
```

Comment:
I have checked what caused the issue with the nan keys, and the main problem is that:

```python
>>> np.nan == np.nan
False
>>> float('nan') == float('nan')
False
```
But also, `-1` could be a valid categorical value, so I'll recommend fill the nan values with `None` wich also represents nullity, but have a more consistent behaviour:

```python
>>> None == None
True
```
I would also recommend make the `fillna` call inside the `get_probability_map` method.